src/NetworkManager.cpp

```diff
@@ -0,0 +1,469 @@
+#include "global.h"
+#include "NetworkManager.h"
+#include "LuaManager.h"
+#include "RageLog.h"
+#include "RageUtil.h"
+#include "StdString.h"
+
+#include <ixwebsocket/IXHttpClient.h>
+#include <ixwebsocket/IXNetSystem.h>
+#include <ixwebsocket/IXUrlParser.h>
+
+#include <algorithm>
+#include <unordered_map>
+#include <vector>
+
+
+NetworkManager*	NETWORK = nullptr;	// global and accessible from anywhere in our program
+
+Preference<bool> NetworkManager::httpEnable("HttpEnable", false);
+Preference<RString> NetworkManager::httpAllowHosts("HttpAllowHosts", "");
+
+NetworkManager::NetworkManager() : httpClient(true)
+{
+	ix::initNetSystem();
+
+	// Register with Lua.
+	{
+		Lua *L = LUA->Get();
+		this->PushSelf(L);
+		lua_setglobal(L, "NETWORK");
+		LUA->Release(L);
+	}
+}
+
+NetworkManager::~NetworkManager()
+{
+	// Unregister with Lua.
+	LUA->UnsetGlobal("NETWORK");
+
+	ix::uninitNetSystem();
+}
+
+bool NetworkManager::IsUrlAllowed(const std::string& url)
+{
+	if (!this->httpEnable.Get())
+	{
+		return false;
+	}
+
+	std::string protocol;
+	std::string host;
+	std::string path;
+	std::string query;
+	int port;
+
+	bool valid = ix::UrlParser::parse(url, protocol, host, path, query, port);
+
+	if (!valid)
+	{
+		return false;
+	}
+
+	if (protocol != "http" && protocol != "https")
+	{
+		return false;
+	}
+
+	vector<RString> allowedHosts;
+	split(this->httpAllowHosts.Get(), ",", allowedHosts);
```

Comment:
Ah okay makes sense! I suppose it would be good to namespace all of this as "split" is very generic, but that would require a lot of change in SM and would be a change for a another day.